src/App.tsx

```diff
@@ -1,223 +1,181 @@
 import { AuthContext } from './components/Auth/AuthContext';
+import { Footer } from './components/Footer/Footer';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Notification } from './components/ErrorNotification/ErrorNotification';
+import { Todo } from './types/Todo';
+import {
+  getTodos,
+  deleteTodo,
+  createTodo,
+  updateTodo,
+} from './api/todos';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [hasError, setHasError] = useState(false);
+  const [filter, setFilter] = useState('all');
+  const [subtitleError, setSubtitleError] = useState('');
+  const [title, setTitle] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState(0);
+
+  if (!hasError) {
+    setTimeout(() => {
+      setHasError(true);
+    }, 3000);
+  }
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    if (user) {
+      getTodos(user.id)
+        .then(response => {
+          setTodos(response);
+          setIsAdding(true);
+        })
+        .catch(() => setHasError(true));
+    }
   }, []);
 
+  const visibleTodos = todos.filter(todo => {
+    switch (filter) {
+      case 'completed':
+        return todo.completed;
+      case 'active':
+        return !todo.completed;
+      default:
+        return todo;
+    }
+  });
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (title.trim().length === 0) {
+      setHasError(true);
+      setSubtitleError('Title can\'t be empty');
+      setIsAdding(true);
+    }
+
+    if (user) {
+      createTodo(user.id, title).then(newTodo => {
+        setTodos([
+          ...todos,
+          newTodo,
+        ]);
+      }).catch(() => {
+        setSubtitleError('Unable to add a todo');
+      });
+    }
+
+    setIsAdding(false);
+    setTitle('');
+  };
+
+  const handleRemove = useCallback(
+    async (todoId: number) => {
+      try {
+        await deleteTodo(todoId);
+        setTodos(prev => prev.filter(todo => todo.id !== todoId));
+      } catch {
+        setHasError(true);
+        setSubtitleError('Unable to delete a todo');
+      }
+    }, [],
+  );
+
+  const handleUpdate = async (todoId: number, data: Partial<Todo>) => {
+    await updateTodo(todoId, data).then((response) => {
+      setTodos(todos.map(todo => (
+        todo.id === todoId
+          ? response
+          : todo
+      )));
+    }).catch(() => {
+      setSubtitleError('Unable to update a todo');
+    });
+  };
+
+  const completedTodos = todos?.filter(todo => todo.completed);
+  const activeToogle = todos.filter(todo => !todo.completed);
```

Comment:
краще ініціалізацію змінних перенести нагору
```suggestion
  const completedTodos = todos?.filter(todo => todo.completed);
  const activeTodos = todos.filter(todo => !todo.completed);
```